src/reducers/shelf/filter.ts

```diff
@@ -0,0 +1,47 @@
+
+import {ADD_ONE_OF_FILTER, ADD_RANGE_FILTER, REMOVE_ONE_OF_FILTER, REMOVE_RANGE_FILTER} from '../../actions/filter';
+import {Action} from '../../actions/index';
+import {DEFAULT_SHELF_UNIT_SPEC, ShelfUnitSpec} from '../../models/shelf/spec';
+import {insertItemToArray, removeItemFromArray} from './spec';
+
+export function filterReducer(shelfSpec: Readonly<ShelfUnitSpec> = DEFAULT_SHELF_UNIT_SPEC,
+                              action: Action): ShelfUnitSpec {
+  switch (action.type) {
+    case ADD_RANGE_FILTER: {
```

Comment:
These two ADD do exactly the same thing, it seems like we only need one `FILTER_ADD`?